### DISCUSSION TITLE
Possible to store into CacheHandler during build (npm run build) ?

### DISCUSSION THREAD

**Discussion by AlexandreBourdeaudhui** [caching-tools/next-shared-cache#436] [Q&A]
Hello, i think no, but i'm trying. 

I want to know if it is possible to store the fetched data, during the build of Nextjs (`yarn run build`), into the Redis CacheHandler ? 

I ask it because when building, it is stored in the filesystem (on disk). Once online, the user triggers the redis cache when visits a page and the data stored in the disk is not longer used ?

FYI: When the app is live, it is on 2 containers/pod. We're using the ISR with a Redis CacheHandler for not building twice on disk (in each container), on (ondemand-)revalidation.

**Comment by better-salmon** [comment]
Hello @AlexandreBourdeaudhui! I apologize for the delay in getting back to you. 

If your Redis server is up and running before the build process, `CacheHandler` will automatically store fetched results in the cache. 

Opting out of caching during the build is a straightforward process. Simply refer [to the documentation section](https://caching-tools.github.io/next-shared-cache/configuration/opt-out-cache-on-build) that outlines the steps. Conversely, to enable caching, it's important not to create the Redis Handler conditionally.

Please let me know if you have any further questions or concerns.

**Reply by AlexandreBourdeaudhui** [reply]
Hello @better-salmon 👋. Thanks for your response. Unfortunately, the `CacheHandler.onCreation` handler seems never called when the build process (with the `npm run build` command). I tried to add a log before any process with Redis, but it's never printed. Do you have a fully example ?
